src/Lightweight/DataMapper/DataMapper.hpp

```diff
@@ -599,13 +595,43 @@ class DataMapper
     ///    // only info.name and info.city are loaded
     /// }
     /// @endcode
-    template <typename ElementMask, typename Record, typename... InputParameters>
+    template <typename lementMask, typename Record, typename... InputParameters>
     std::vector<Record> Query(SqlSelectQueryBuilder::ComposedQuery const& selectQuery,
                               InputParameters&&... inputParameters);
 
+    /// Queries records of different types from the database, based on the given query.
+    /// User can constructed query that selects columns from the multiple tables
+    /// this function is uset to get result of the
+    ///
+    /// example:
+    /// @code
+    ///
+    /// struct JointA{};
+    /// struct JointB{};
+    /// struct JointC{};
+    ///
+    /// // the following query will construct statement to fetch all elements of JointA and JointC types
+    /// auto dm = DataMapper {};
+    /// const auto query = dm.FromTable(RecordTableName<JoinTestA>)
+    ///             .Select()
+    ///             .Fields<JointA, JointC>()
+    ///             .InnerJoin<&JointB::a_id, &JointA::id>()
+    ///             .InnerJoin<&JointC::id, &JointB::c_id>()
+    ///             .All();
+    /// const auto records= dm.Query<JointA, JointC>(query);
```

Comment:
```suggestion
    /// const auto records = dm.Query<JointA, JointC>(query);
```
Ok, lol. My vision scope today is below zero, lol. But at least let's have proper formatting :D
And also suggest slightly more real-world examples, like:

```suggestion
    /// const auto records = dm.Query<JointA, JointC>().Where....All(); // with correct calls ofc
```